src/model.ts

```diff
@@ -405,10 +421,51 @@ export class JSORMBase {
   }
   set isPersisted(val: boolean) {
     this._persisted = val
-    this.reset()
+    if (!!val) this.reset()
+  }
+
+  _onStoreChange: Function
+  private onStoreChange() : Function {
+    if (this._onStoreChange) return this._onStoreChange
+    this._onStoreChange =  (_event: any, attrs: any) => {
+      Object.keys(attrs).forEach((k) => {
+        if (this[k] !== attrs[k]) this[k] = attrs[k]
+      })
+    }
+    return this._onStoreChange
+  }
+
+  unlisten() {
```

Comment:
Would mark these two as `void` return types. Could see this accidentally evolving somehow.